src/Microsoft.AspNetCore.Mvc.Analyzers/DiagnosticDescriptors.cs

```diff
@@ -52,5 +52,14 @@ public static class DiagnosticDescriptors
                 DiagnosticSeverity.Warning,
                 isEnabledByDefault: true,
                 helpLinkUri: "https://aka.ms/AA20pbc");
+
+        public static readonly DiagnosticDescriptor MVC1005_ParameterAttributeAvoidUsingRequiredOncollections =
+            new DiagnosticDescriptor(
+                "MVC1005",
+                "Use of 'RequiredAttribute' should be avoided on collection",
+              "Use of '{0}' should be avoided. This always made model validation is true. Consider using [MinLength(1)].",
```

Comment:
This advice works only for arrays in .NET Framework projects. Even in .NET Core, this advice doesn't work for `IEnumerable<T>` but `IEnumerable<T>` has the same issues w/ `[Required]`. Probably need an additional diagnostic without this advice for the excluded cases.